src/jobflow_remote/jobs/jobcontroller.py

```diff
@@ -2755,8 +2840,14 @@ def add_flow(
         # should not lead to inconsistencies in the states, even if one of
         # the jobs is checked out in the meanwhile. The opposite could lead
         # to errors.
-        self.flows.insert_one(flow_doc)
-        self.jobs.insert_many(job_dicts)
+        try:
+            self.flows.insert_one(flow_doc)
+            self.jobs.insert_many(job_dicts)
```

Comment:
If the flow can be inserted but not one of the jobs, we end up with one flow in the db without its jobs. Correct ? Could this be problematic ?

Reply:
yes, that is one case that I thought about. However this cannot happen because an issue in the submission script. 
```python
j = add(1,2)

for i in range(5):
    flow = Flow([j])
    submit_flow(flow)
```
This script will not even get to `submit_flow` in the second loop, because jobflow will directly prevent to insert the same Job in two different Flows.
I imagined that the only possible occurrence of an inconsistent state is if there is a random clash of uuid. (or if a user deliberately sets the uuid in two jobs to be equal). Can you foresee other cases where the flow uuid is different and one of the job ids is duplicated?

In that case I would really need to manually check the db for clashes or perform all the action as a MongoDB transaction. The latter might not be so bed for this action. And it is not something done in the runner that will take much time. I would maybe need to make some tests to see if this can get much slower. What do you think?

Reply:
Indeed, I forgot about jobflow already making this check. I then don't see such a case unless with a random clash or explicit uuid setting. Then it's probably not needed to do the transaction or the manual check. I would add a comment there to explain why we don't need this check (maybe some test should be added to make sure jobflow continues to behave this way ? not sure if it's needed, probably not..).